qemu/tests/libvirt_host_model_test.py

```diff
@@ -0,0 +1,175 @@
+        exclude_map = eval(params.get('exclude_map', '{}'))
+        check_exclude = False
+        if cpu_model_qemu in exclude_map.keys():
+            exclude_map_flags = exclude_map[cpu_model_qemu]
+            check_exclude = True
+        miss_flags = []
+        mismatch_flags = []
+        result_bool = True
+        for flag in libvirt_cpu_flags.keys():
+            if flag not in qemu_cpu_flags.keys():
+                if libvirt_cpu_flags[flag] == 'on':
+                    miss_flags.append(flag)
+            elif libvirt_cpu_flags[flag] != qemu_cpu_flags[flag]:
+                mismatch_flags.append(flag)
+        if miss_flags:
+            logging.error("\nmiss flags %s from qemu cli\n", miss_flags)
+            for miss_flag in miss_flags:
+                if check_exclude and miss_flag not in exclude_map_flags:
+                    result_bool = False
+                    break
+        if mismatch_flags:
+            logging.error("\nmismatch flags %s between libvirt and qemu\n",
+                          mismatch_flags)
+            for mismatch_flag in miss_flags:
+                if check_exclude and mismatch_flag not in exclude_map_flags:
+                    result_bool = False
+                    break
+        return result_bool
+    finally:
+        if virsh.is_alive(name):
+            virsh.destroy(name, ignore_status=True)
+        virsh.undefine(name, ignore_status=True)
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test cpu models with full flags that libvirt generates
+    1) start a default configuration xml by virsh
+    2) get the cpu elements from libvirt xml by 'virsh dumpxml $domain'
+    3) parse the cpu_flags and store into libvirt_cpu_flags
+    4) get the cpu flags from qemu process and stroe to qemu_cpu_flags
+    5) Compare the cpu flags, model of xml and qemu process, should be same
+    6) Boot guest with combined flags, model, guest works well
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    if not compare_cpu_info(test, params):
+        logging.error("\ncpu libvirt flags are %s\n",
+                      (str(params["libvirt_cpu_info"])))
+        logging.error("\ncpu qemu cli flags are %s\n",
+                      (str(params["qemu_cpu_info"])))
+        test.fail("CPU info is different between dumpxml and qemu cli")
+
+    cpu_flags = params.get("cpu_model_flags")
+    params["cpu_model_flags"] = cpu.recombine_qemu_cpu_flags(
+        params["qemu_proc_cpu_flags"], cpu_flags)
+    params["start_vm"] = "yes"
+    vm_name = params['main_vm']
+    env_process.preprocess_vm(test, params, env, vm_name)
+
+    vm = env.get_vm(vm_name)
+    error_context.context("Try to log into guest", logging.info)
+    vm.wait_for_login()
+
+    if params.get("reboot_method"):
+        error_context.context("Reboot guest '%s'." % vm.name, logging.info)
+        vm.reboot()
```

Comment:
For this case,  L171~L173 are dead code,  and also no this step in the doc.  Please update it according to your test requirement.

Reply:
update and thanks